src/views/Students/StudentComponents/AlumniServiceForm.jsx

```diff
@@ -39,14 +39,27 @@ const AlumniServiceForm = (props) => {
   const [receiptNumberValue, setReceiptNumberValue] = useState(props.alumniService ? props.alumniService.receipt_number : '');
   const [validationRules, setValidationRules] = useState(AlumniServiceValidations);
   const [feeFieldsRequired, setFeeFieldsRequired] = useState(false);
-  const [selectedCategory, setSelectedCategory] = useState('');
   const [peerLearningOptions, setPeerLearningOptions] = useState([]);
   const [categoryOptions, setCategoryOptions] = useState([]);
   const [youthLeadershipOptions, setYouthLeadershipOptions] = useState([]);
   const [programOptions, setProgramOptions] = useState([]);
   const [ideationClubOptions, setIdeationClubOptions] = useState([]);
   const [givebackOptions , setGivebackOptions] = useState([]);
 
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [selectedSubCategory, setSelectedSubCategory] = useState('');
+
+  const [selectedYouthLeadership, setSelectedYouthLeadership] = useState('');
+  const [selectedPeerLearning, setSelectedPeerLearning] = useState('');
+  const [selectedIdeationClub, setSelectedIdeationClub] = useState('');
+  const [selectedGiveback, setSelectedGiveback] = useState('');
+
+  useEffect(() => {
+    if (props.alumniService) {
+      setSelectedCategory(props.alumniService ? props.alumniService.category : "");
+    }
+  },[props.alumniService,show]);
```

Comment:
any reason for putting `show` here?

Reply:
when the form appears then the saved value of selectedCategory will be reset to its initial value 
which then the display the next field based on it.